pandas/tests/tools/test_to_datetime.py

```diff
@@ -1489,6 +1489,29 @@ def test_to_datetime_fixed_offset(self):
 
 
 class TestToDatetimeUnit:
+    @pytest.mark.parametrize("unit", ["Y", "M"])
+    def test_to_datetime_month_or_year_unit_non_round_float(self, cache, unit):
+        # GH#50301
+        # Match Timestamp behavior in disallowing non-round floats with
+        #  Y or M unit
+        msg = f"Conversion of non-round float with unit={unit} is ambiguous"
+        with pytest.raises(ValueError, match=msg):
+            to_datetime([1.5], unit=unit, errors="raise")
+
+        # with errors="ignore" we also end up raising within the Timestamp
+        #  constructor; this may not be ideal
+        with pytest.raises(ValueError, match=msg):
+            to_datetime([1.5], unit=unit, errors="ignore")
+
+        res = to_datetime([1.5], unit=unit, errors="coerce")
+        expected = Index([NaT], dtype="M8[ns]")
+        tm.assert_index_equal(res, expected)
+
+        # round floats are OK
+        res = to_datetime([1.0], unit=unit)
+        expected = to_datetime([1], unit=unit)
```

Comment:
can this be constructed explicitly with `Index` / `DatetimeIndex` as you do above?

Reply:
i find this a lot simpler

Reply:
same, but isn't it risk to have both `res` and `expected` go down nearly exactly the same path?

Reply:
the point of this test is that they behave the same.  some other test can check that the integer case behaves "correctly"

Reply:
ok sure